test/db.tests.js

```diff
@@ -1074,11 +1180,250 @@ describe('LimitDBRedis', () => {
+            .then((ttl) => assert.equal(ttl, expectedTTL))
+            .then(() => done());
+        });
+
+        it('should decrease erlQuota when we activate ERL', (done) => {
+          params.erlQuota = {key: 'erlquotakey', per_cal_month: 10};
+
+          // check erl not activated yet
+          redisExistsPromise(erlIsActiveKey)
+            .then((erlIsActiveExists) => assert.equal(erlIsActiveExists, 0))
+            // check erlQuotaKey does not exist
+            .then(() => redisExistsPromise(params.erlQuota.key)
+              .then((erlQuotaKeyExists) => assert.equal(erlQuotaKeyExists, 0)))
+            // attempt to take elevated should work for first token
+            .then(() => takeElevatedPromise(params))
+            .then(() => redisExistsPromise(erlIsActiveKey))
+            .then((erlIsActiveKeyExists) => assert.equal(erlIsActiveKeyExists, 0))
+            // erl now activated
+            .then(() => takeElevatedPromise(params))
+            .then(() => redisExistsPromise(erlIsActiveKey))
+            .then((erlIsActiveKeyExists) => assert.equal(erlIsActiveKeyExists, 1))
+            // check erlQuota should be decreased by 1
+            .then(() => redisGetPromise(params.erlQuota.key))
+            .then((erlQuotaKeyValue) => assert.equal(erlQuotaKeyValue, params.erlQuota.per_cal_month - 1))
+            .then(() => done());
+        });
+
+        it('should not activate ERL when erlQuota is 0', (done) => {
+          params.erlQuota = {key: 'erlquotakey', per_cal_month: 0};
+
+          // check erl not activated yet
+          redisExistsPromise(erlIsActiveKey)
+            .then((erlIsActiveExists) => assert.equal(erlIsActiveExists, 0))
+            // check erlQuotaKey does not exist
+            .then(() => redisExistsPromise(params.erlQuota.key)
+              .then((erlQuotaKeyExists) => assert.equal(erlQuotaKeyExists, 0)))
+            // attempt to take elevated should work for first token
+            .then(() => takeElevatedPromise(params))
+            .then((result) => assert.isTrue(result.conformant) && assert.isFalse(result.erl_activated))
+            .then(() => redisExistsPromise(erlIsActiveKey))
+            .then((erlIsActiveKeyExists) => assert.equal(erlIsActiveKeyExists, 0))
+            // next takeElevated should have attempted to activate ERL but failed
+            .then(() => takeElevatedPromise(params))
+            .then((result) => assert.isFalse(result.conformant) && assert.isFalse(result.erl_activated))
+            .then(() => redisExistsPromise(erlIsActiveKey))
+            .then((erlIsActiveKeyExists) => assert.equal(erlIsActiveKeyExists, 0))
+            // check erlQuota was not set
+            .then(() => redisGetPromise(params.erlQuota.key))
+            .then((erlQuotaKeyValue) => assert.isNull(erlQuotaKeyValue, 0))
+            .then(() => done());
+        });
+
+        it('should activate ERL when erlQuota is 1', (done) => {
```

Comment:
```suggestion
        it('should activate ERL when erlQuota is not 0', (done) => {
```


Also `should decrease erlQuota when we activate ERL` already accounts for this test case, you can remove it if you want

Reply:
> Also should decrease erlQuota when we activate ERL already accounts for this test case, you can remove it if you want
You're right. I was testing the border case of quota=1 not setting `quota-1` in Redis and then interpreting there was no quota left.
But it's a special case of `should decrease erlQuota when we activate ERL`, so I'll change the quota in that test to 1.

Reply:
I see you've done that change already - do you want to keep this test still?